### PR TITLE
system variables: add two TSO related variables

### PR DESCRIPTION
**Pull request by en-jin19**
### What is changed, added or deleted? (Required)
Add TSO related variables `tidb_enable_tso_follower_proxy` and `tidb_tso_client_batch_max_wait_time`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:

1. For the description of newly added variables, please refer to https://github.com/pingcap/docs-cn/pull/7364.
2. For the order of newly added variables in the document, please refer to https://github.com/pingcap/docs-cn/pull/7432.

- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -819,6 +819,19 @@ Query OK, 0 rows affected (0.09 sec)
 - Default value: `ON`
 - This variable is used to dynamically control whether the telemetry collection in TiDB is enabled. By setting the value to `OFF`, the telemetry collection is disabled. If the [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-new-in-v402) TiDB configuration item is set to `false` on all TiDB instances, the telemetry collection is always disabled and this system variable will not take effect. See [Telemetry](/telemetry.md) for details.
 
+### tidb_enable_tso_follower_proxy <span class="version-mark">New in v5.3</span>
+
+- Scope: GLOBAL
+- Default value: `OFF`
+- This variable is used to enable the TSO Follower Proxy feature. When the value is `OFF`, TiDB only gets TSO from PD leaders. After this feature is enabled, TiDB gets TSO by evenly sending requests to all PD nodes and forwarding TSO requests through PD followers. By that means, the CPU pressure of PD leaders is reduced.
+- Scenarios for enabling TSO Follower Proxy:
+    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, and this issue causes high latency of TSO RPC requests.
```
Comment (JmPotato) [inline system-variables.md:828]:
```suggestion
    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.
```

File: system-variables.md [hunk 2]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
...
 
+### tidb_tso_client_batch_max_wait_time <span class="version-mark">New in v5.3</span>
+
+- Scope: GLOBAL
+- Default value: `0`
+- Range: `[0, 10]`
+- Unit: Milliseconds
+- This variable is used to set the maximum waiting time for a batch operation when TiDB requests TSO from PD. The default value is `0`, which means no extra waiting is performed.
+- When obtaining TSO requests from PD, PD client, used by TiDB, collects as many TSO requests that are received at the same time as possible at one time. Then, PD client consolidates those requests into one RPC request and sends them to PD, thereby reducing the pressure on PD.
+- After setting this value other than 0, TiDB waits for the maximum duration of this value before the end of each batch. This is to collect more TSO requests, thereby improving the batching effect.
```
Comment (JmPotato) [inline system-variables.md:1524]:
```suggestion
- After setting this value greater than 0, TiDB waits for the maximum duration of this value before the end of each batch. This is to collect more TSO requests, thereby improving the batching effect.
```

File: system-variables.md [hunk 3]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
...
+- Scenarios for increasing the value of this variable:
+    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, and this issue causes high latency of TSO RPC requests.
```
Comment (JmPotato) [inline system-variables.md:1526]:
```suggestion
    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.
```

**Review by JmPotato — APPROVED** [review]
The rest LGTM.

File: system-variables.md [hunk 4]
```diff
@@ -819,6 +819,19 @@ Query OK, 0 rows affected (0.09 sec)
 - Default value: `ON`
 - This variable is used to dynamically control whether the telemetry collection in TiDB is enabled. By setting the value to `OFF`, the telemetry collection is disabled. If the [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-new-in-v402) TiDB configuration item is set to `false` on all TiDB instances, the telemetry collection is always disabled and this system variable will not take effect. See [Telemetry](/telemetry.md) for details.
 
+### tidb_enable_tso_follower_proxy <span class="version-mark">New in v5.3</span>
+
+- Scope: GLOBAL
+- Default value: `OFF`
+- This variable is used to enable the TSO Follower Proxy feature. When the value is `OFF`, TiDB only gets TSO from PD leaders. After this feature is enabled, TiDB gets TSO by evenly sending requests to all PD nodes and forwarding TSO requests through PD followers. By that means, the CPU pressure of PD leaders is reduced.
```
Comment (TomShawn) [inline system-variables.md:826]:
```suggestion
- This variable is used to enable the TSO Follower Proxy feature. When the value is `OFF`, TiDB only gets TSO from the PD leader. After this feature is enabled, TiDB gets TSO by evenly sending requests to all PD nodes and forwarding TSO requests through PD followers. This helps reduce the CPU pressure of PD leader.
```

File: system-variables.md [hunk 5]
```diff
@@ -819,6 +819,19 @@ Query OK, 0 rows affected (0.09 sec)
...
+- Scenarios for enabling TSO Follower Proxy:
+    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.
```
Comment (TomShawn) [inline system-variables.md:828]:
```suggestion
    * Due to the high pressure of TSO requests, the CPU of the PD leader reaches a bottleneck, which causes high latency of TSO RPC requests.
```

File: system-variables.md [hunk 6]
```diff
@@ -819,6 +819,19 @@ Query OK, 0 rows affected (0.09 sec)
...
+    * The TiDB cluster has a large number of TiDB instances, and increasing the value of [tidb_tso_client_batch_max_wait_time](#tidb_tso_client_batch_max_wait_time-new-in-v53) cannot alleviate the issue of high latency of TSO RPC requests.
```
Comment (TomShawn) [inline system-variables.md:829]:
```suggestion
    * The TiDB cluster has many TiDB instances, and increasing the value of [`tidb_tso_client_batch_max_wait_time`](#tidb_tso_client_batch_max_wait_time-new-in-v53) cannot alleviate the high latency issue of TSO RPC requests.
```

File: system-variables.md [hunk 7]
```diff
@@ -819,6 +819,19 @@ Query OK, 0 rows affected (0.09 sec)
...
+
+> **Note:**
+>
+> Suppose the TSO RPC latency increases, but the CPU bottleneck of PD leaders does not cause this issue. In this case, enabling the TSO Follower Proxy might increase the latency of TiDB statement execution, thereby affecting the QPS performance of the cluster.
```
Comment (TomShawn) [inline system-variables.md:833]:
```suggestion
> Suppose that the CPU of the PD leader does not reach a bottleneck and the TSO RPC latency does not increase. In this case, enabling the TSO Follower Proxy might increase the execution latency in TiDB and affect the QPS performance of the cluster.
```

File: system-variables.md [hunk 8]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
 - Range: `[0, 9223372036854775807]`
 - This variable is used to limit the maximum number of requests TiDB can send to TiKV at the same time. 0 means no limit.
 
+### tidb_tso_client_batch_max_wait_time <span class="version-mark">New in v5.3</span>
+
+- Scope: GLOBAL
+- Default value: `0`
+- Range: `[0, 10]`
+- Unit: Milliseconds
+- This variable is used to set the maximum waiting time for a batch operation when TiDB requests TSO from PD. The default value is `0`, which means no extra waiting is performed.
```
Comment (TomShawn) [inline system-variables.md:1522]:
```suggestion
- This variable is used to set the maximum waiting time for a batch operation when TiDB requests TSO from PD. The default value is `0`, which means no extra waiting time.
```

File: system-variables.md [hunk 9]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
...
+- When obtaining TSO requests from PD, PD client, used by TiDB, collects as many TSO requests that are received at the same time as possible at one time. Then, PD client consolidates those requests into one RPC request and sends them to PD, thereby reducing the pressure on PD.
```
Comment (TomShawn) [inline system-variables.md:1523]:
```suggestion
- When obtaining TSO requests from PD each time, PD Client, used by TiDB, collects as many TSO requests received at the same time as possible. Then, PD Client merges the collected requests in batch into one RPC request and sends the request to PD. This helps reduce the pressure on PD.
```

File: system-variables.md [hunk 10]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
...
+- After setting this value greater than 0, TiDB waits for the maximum duration of this value before the end of each batch. This is to collect more TSO requests, thereby improving the batching effect.
```
Comment (TomShawn) [inline system-variables.md:1524]:
```suggestion
- After setting this variable to a value greater than `0`, TiDB waits for the maximum duration of this value before the end of each batch merge. This is to collect more TSO requests and improve the effect of batch operations.
```

File: system-variables.md [hunk 11]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
...
+- Scenarios for increasing the value of this variable:
+    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.
+    * There are not many TiDB instances in the cluster, but every TiDB instance is under relatively high concurrency.
```
Comment (TomShawn) [inline system-variables.md:1527]:
```suggestion
    * There are not many TiDB instances in the cluster, but every TiDB instance is in high concurrency.
```

File: system-variables.md [hunk 12]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
...
+- When setting this variable, it is recommended to set it to a smaller value as much as possible.
+
+> **Notes:**
+>
+> Suppose the TSO RPC latency increases, but the CPU bottleneck of PD leaders does not cause this issue. In this case, increasing the value of `tidb_tso_client_batch_max_wait_time` might increase the latency of TiDB statement execution, thereby affecting the QPS performance of the cluster.
```
Comment (TomShawn) [inline system-variables.md:1532]: ditto

File: system-variables.md (hunk 7 above)
Comment (en-jin19) [reply system-variables.md:833]: 这部分的第一句改成这样如何呢？
Suppose that the CPU of the PD leader does not reach a bottleneck but the TSO RPC latency increases.

File: system-variables.md [hunk 13]
```diff
@@ -1500,6 +1513,24 @@ SET tidb_slow_log_threshold = 200;
...
+
+- Scope: GLOBAL
+- Default value: `0`
+- Range: `[0, 10]`
+- Unit: Milliseconds
+- This variable is used to set the maximum waiting time for a batch operation when TiDB requests TSO from PD. The default value is `0`, which means no extra waiting is performed.
+- When obtaining TSO requests from PD, PD client, used by TiDB, collects as many TSO requests that are received at the same time as possible at one time. Then, PD client consolidates those requests into one RPC request and sends them to PD, thereby reducing the pressure on PD.
+- After setting this value greater than 0, TiDB waits for the maximum duration of this value before the end of each batch. This is to collect more TSO requests, thereby improving the batching effect.
+- Scenarios for increasing the value of this variable:
+    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.
```
[the remark this answers is not in the document]
Comment (en-jin19) [reply system-variables.md:1526]:
```suggestion
    * Due to the high pressure of TSO requests, the CPU of the PD leader reaches a bottleneck, which causes high latency of TSO RPC requests.
```

File: system-variables.md (hunk 7 above)
Comment (TomShawn) [reply system-variables.md:833]: @JmPotato 【如果 PD leader 未因 CPU 限制达到瓶颈而导致 TSO RPC 延迟升高】有歧义。这里是说 TSO RPC 延迟没有升高吗？

Comment (en-jin19) [reply system-variables.md:833]: 是不是说 TSO RPC 延迟升高了但是这个升高不是因为 PD leader CPU 达到瓶颈而导致的呢？

Comment (en-jin19) [reply system-variables.md:833]: 刚刚确认到是指： TSO gRPC 的延迟升高不是因为 PD leader CPU 达到瓶颈（是由网络等问题导致的）

Comment (TomShawn) [reply system-variables.md:833]: @en-jin19 谢谢，能辛苦根据“TSO gRPC 的延迟升高不是因为 PD leader CPU 达到瓶颈（是由网络等问题导致的）”这一句修改下中文吗？🙏

File: system-variables.md (hunk 12 above)
Comment (en-jin19) [reply system-variables.md:1532]:
```suggestion
> Suppose that the CPU of the PD leader does not reach a bottleneck but the TSO RPC latency increases for other reasons (such as network issues). In this case, increasing the value of `tidb_tso_client_batch_max_wait_time` might increase the execution latency in TiDB and affect the QPS performance of the cluster.
```

File: system-variables.md [hunk 14]
```diff
@@ -819,6 +819,19 @@ Query OK, 0 rows affected (0.09 sec)
...
+- Scope: GLOBAL
+- Default value: `OFF`
+- This variable is used to enable the TSO Follower Proxy feature. When the value is `OFF`, TiDB only gets TSO from the PD leader. After this feature is enabled, TiDB gets TSO by evenly sending requests to all PD nodes and forwarding TSO requests through PD followers. This helps reduce the CPU pressure of PD leader.
+- Scenarios for enabling TSO Follower Proxy:
+    * Due to the high pressure of TSO requests, the CPU of the PD leader reaches a bottleneck, which causes high latency of TSO RPC requests.
+    * The TiDB cluster has many TiDB instances, and increasing the value of [`tidb_tso_client_batch_max_wait_time`](#tidb_tso_client_batch_max_wait_time-new-in-v53) cannot alleviate the high latency issue of TSO RPC requests.
+
+> **Note:**
+>
+> Suppose that the CPU of the PD leader does not reach a bottleneck but the TSO RPC latency increases for other reasons (such as network issues). In this case, enabling the TSO Follower Proxy might increase the execution latency in TiDB and affect the QPS performance of the cluster.
```
Comment (en-jin19) [inline system-variables.md:833]:
```suggestion
> Suppose that the TSO RPC latency increases for reasons other than a CPU bottleneck of the PD leader (such as network issues). In this case, enabling the TSO Follower Proxy might increase the execution latency in TiDB and affect the QPS performance of the cluster.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**